CHANGELOG.md

```diff
@@ -37,20 +40,24 @@
     }
   });
   ```
+* Added support for realm list of nullable primitive types, ie. `RealmList<int?>`. ([#163](https://github.com/realm/realm-dart/issues/163))
+* Allow null arguments on query. ([#871](https://github.com/realm/realm-dart/issues/871))
 
 ### Fixed
-* Allow null arguments on query. ([#871](https://github.com/realm/realm-dart/issues/871))
 * Previously removeAt did not truncate length. ([#883](https://github.com/realm/realm-dart/issues/883))
-* List.length= now throws, if you try to increase length, ([#894](https://github.com/realm/realm-dart/pull/894)).
+* List.length= now throws, if you try to increase length. This previously succeeded silently. ([#894](https://github.com/realm/realm-dart/pull/894)).
 * Queries on lists were broken. ([#909](https://github.com/realm/realm-dart/issues/909))
+  Example: 
+  ```dart
+  expect(realm.all<Person>(), [alice, bob, carol, dan]); // assume this pass, then ...
+  expect(team.players.query('TRUEPREDICATE'), [alice, bob]); // <-- ... this fails and return the same as realm.all<Person>()
+  ```
 * Queries on results didn't filter the existing results. ([#908](https://github.com/realm/realm-dart/issues/908)).
-  As an example
+  Example
   ```dart
-  expect(realm.query<Person>('FALSEPREDICATE').query('TRUEPREDICATE'), isEmpty);
+  expect(realm.query<Person>('FALSEPREDICATE').query('TRUEPREDICATE'), isEmpty); //<-- Fails if a Persion object exists
   ```
-  would fail if any Persons exists
-* Fixed an issue that would cause passwords sent to the server (e.g. `Credentials.EmailPassword` or `EmailPasswordAuthProvider.registerUser`) to contain an extra empty byte at the end. (PR [#918](https://github.com/realm/realm-dart/pull/918))
-* Added support for realm list of nullable primitive types, ie. `RealmList<int?>`. ([#163](https://github.com/realm/realm-dart/issues/163))
+* Fixed copying of native structs for session errors and http requests. ([#924](https://github.com/realm/realm-dart/issues/924))
```

Comment:
Maybe it will be nice to explain here in the Changelog what the fixed problem was. Did we have an error C3490 or something else? Do we have some issue that had to be attached? This link ([#924](https://github.com/realm/realm-dart/issues/924)) point to the same PR. Reading the PR I don't understand why we need this change unles I following the comments.

Reply:
its mostly an internal fix. not sure how can we improve the messaging here. It is fine to link to the PR in the changelog and we do it in many places. We don't have an issue for this fix.

Reply:
Updated the PR description

Reply:
OK. I thought the link was broken because it pointed to the issues 'realm-dart/issues/924' instead of to the pulls 'realm-dart/pull/924'. It is still working, but should be 'pull'.

Reply:
that's good point. will fix